src/osyris/core/vector.py

```diff
@@ -252,3 +252,65 @@ def cross(self, other):
         z = self.x * other.y
         z -= self.y * other.x
         return self.__class__(x, y, z)
+
+    def _get_parent_pos(self):
+        group = self.parent
+        return group.get("position", group.parent["amr"]["position"])
+
+    def _get_colatitude(self, pos):
+        colatitude = np.arctan2(np.linalg.norm([pos.y.values, pos.x.values], axis=0),
+                                pos.z.values)
+        return Array(values=colatitude, unit="rad", name=pos.name + "_theta")
+
+    def _get_azimuth(self, pos):
+        azimuth = np.arctan2(pos.y.values, pos.x.values)
+        return Array(values=azimuth, unit="rad", name=pos.name + "_phi")
+
+    @property
+    def r(self):
+        if self.unit.is_compatible_with("meter"):
```

Comment:
At first, I thought this is going to break down somewhere, but after thinking about it, maybe it's not a bad idea. I could only think of a quantity representing a wavelength, for which this would go wrong. And I don't think this is very common in Ramses? Although maybe with radiation and multigroup?

Reply:
I think it should be up to the user to decide whether calling a spherical/cylindrical coordinate computation on their vector makes sense physically or not. We can't predict all the possible lengths the user might be working with, so it's best we leave it up to them